### PR TITLE
Added timeout for purge requests

### PR DESCRIPTION
**Pull request by tomasinchoo**

### CODE REVIEW

File: src/Api.php [hunk 1]
```diff
@@ -289,6 +289,7 @@ protected function query($uri, array $data = [], $method = 'GET', array $headers
           return $this->httpClient->post($this->host . $uri, $data);
 
         case 'PURGE':
+          $data['connect_timeout'] = 3;
```
Comment (greylabel) [inline src/Api.php]: @tomasinchoo Thanks for this PR!

Rather then specifying `3` here directly, I think this approach could be improved in one of two ways:

1. Use a service parameter to allow for end-user overrides of this value. We current do this for `fastly.host`, for example.

Defined here: https://github.com/vvuksan/fastly-drupal/blob/8.x-3.x/fastly.services.yml#L2
Passed to the service: https://github.com/vvuksan/fastly-drupal/blob/8.x-3.x/fastly.services.yml#L7
Used in the service: https://github.com/vvuksan/fastly-drupal/blob/8.x-3.x/src/Api.php#L59

2. Define a class constant to provide more clarity. Something like:

`const FASTLY_API_CONNECT_TIMEOUT = 3;`

Also, do we want to apply this to POST requests as well, @vvuksan

Comment (vvuksan) [reply src/Api.php]: I suppose we should apply timeout to surrogate key requests as well

Comment (tomasinchoo) [reply src/Api.php]: Thanks for feedback. I make changes early in the morning.

Comment (greylabel) [reply src/Api.php]: @tomasinchoo Cool. I think either approach should be fine.

**Comment by tomasinchoo** [comment]
Hi everyone,
I added connect timeout as service parameter.

File: src/Api.php [hunk 2]
```diff
@@ -49,13 +56,21 @@ class Api {
    * @param \Drupal\fastly\State $state
    *   Fastly state service for Drupal.
    */
-  public function __construct(ConfigFactoryInterface $config_factory, $host, ClientInterface $http_client, LoggerInterface $logger, State $state) {
+  public function __construct(ConfigFactoryInterface $config_factory,
```
Comment (greylabel) [inline src/Api.php]: We should verify this indentation meets Drupal coding standards.

**Comment by greylabel** [comment]
@tomasinchoo Looking good overall! Added one additional comment.

File: src/Api.php (hunk 2 above)
Comment (tomasinchoo) [reply src/Api.php]: On it!

Comment (tomasinchoo) [reply src/Api.php]: I've made changes how this constructor declaration is formated.

**Review by greylabel — COMMENTED** [review]
LGTM from a code perspective.

**Pull request merged**